### PR TITLE
Unit Test Cases for Non_Existence_Element

### PR DESCRIPTION
**Pull request by mcalderaz**

### CODE REVIEW

**Comment by jlecount-sungevity** [comment]
- lines 171,172: Please use doublequotes for doc strings
- please put comments before the blocks they describe
- please delete lines 55-58

File: tests/null_object_tests.py [hunk 1]
```diff
@@ -87,3 +95,99 @@ class SimplePage(Page):
         self.driver.refresh()
         with hiro.Timeline().scale(10):
             self.assertEqual(page.id_el_changed, NonexistentElement())
+
+    # test object equivalence/nonequivalence test that expected parameters are displayed
+
+    def test_non_existence_eq_element(self):
```
Comment (mcalderaz) [inline tests/null_object_tests.py]: The test for object equivalence should test that our overloaded __eq__method evaluates the way we expect it to for objects of the same class.  E.g:
x = NonexistentElement()
y = NonexistentElement('value1', 'value2', 'value3')
assert(x.property1 != y.property1) #Objects have different property values
assert x == y  #Our overloaded method should still return true
z = 1
assert x !=z  #An Int type, or any other Object that's not a NonexistentElement type, should return False

**Comment by jlecount-sungevity** [comment]
why are you commenting on your own PR? ;-)

File: tests/null_object_tests.py [hunk 2]
```diff
@@ -87,3 +95,99 @@ class SimplePage(Page):
...
+        first_el_text = non_exist_el.__ne__(page.class_el.text)
+        second_el_text = non_exist_el.__ne__(page.class_xpath_el.text)
+
+        # print first_el_text == second_el_text
+        self.assertEqual(first_el_text != second_el_text, False,
+                         "NonExistence __ne__ unit test has failed")
+
+        # test when str() is called on the object test that expected parameters are displayed
+
+    def test_non_existence_str_element(self):
```
Comment (mcalderaz) [inline tests/null_object_tests.py]: To test our implementation of **str** and **repr** you should use global methods str() and repr() like the following:
x = NonexistentElement('val1', 'val2', 'val3')
toStr = str(x)
#do some assertions on toStr based on what you initialized object 'x' with.
#Should be a similar test for **repr**

File: tests/null_object_tests.py [hunk 3]
```diff
@@ -87,3 +95,99 @@ class SimplePage(Page):
...
+        non_exist_el = NonexistentElement()
+
+        try:
+            result = repr(NonexistentElement(page.exception_msg, page.class_el.text, page.query_string))
+            self.assertIsNotNone(result,"NonExistence __repr__ unit test has failed")
+        except (WebDriverException) as exception:
+            assert exception.__class__.__name__ == "simple_class"
+            raise exception
+
+    '''test that if an undefined property is referenced, an Exception is thrown that includes the initialization data
```
Comment (mcalderaz) [inline tests/null_object_tests.py]: To test this functionality you only need to try accessing an undefined property on the object. e.g:
x = NonexistentElement('val1', 'val2', 'val3')
try:
  x.propThatDoesn'tExist = 0
except Exception as e:
   #Do some assertions against e

File: tests/null_object_tests.py [hunk 4]
```diff
@@ -44,10 +52,10 @@ class SimplePage(Page):
 
         uri = make_temp_page(ElementTest.page_content)
         page = SimplePage(self.driver, uri)
-        try:
-            x=page.invalid_el.text
-        except Exception as e:
-            assert "NoSuchElementException" in e.message
+       # try:
```
Comment (mcalderaz) [inline tests/null_object_tests.py]: Please remove commented out code

**Pull request merged**